File: src/solution_java/easy/_661ImageSmoother.java
package solution_java.easy;

public class _661ImageSmoother {
    public class Solution {
        public int[][] imageSmoother(int[][] img) {
            int row = img.length;
            int column = img[0].length;
            int[][] result = new int[row][column];
            for (int i = 0; i < row; i++) {
                for (int j = 0; j < column; j++) {
                    int count = 0;
                    int sum = 0;
                    for (int k = i - 1; k <= i + 1; k++) {
                        for (int l = j - 1; l <= j + 1; l++) {
                            if (k >= 0 && k < row && l >= 0 && l < column) {
                                count += 1;
                                sum += img[k][l];
                            }
                        }
                    }
                    result[i][j] = sum / count;
                }
            }
            return result;
        }
    }
}

File: src/easy/_661ImageSmoother.kt
package easy

class _661ImageSmoother {
    class Solution {
        fun imageSmoother(img: Array<IntArray>): Array<IntArray> {
            val row = img.size
            val column = img.first().size
            val result = Array(row) { IntArray(column) }
            for (i in 0 until row) {
                for (j in 0 until column) {
                    var count = 1
                    var sum = img[i][j]
                    if (j - 1 >= 0) {
                        count += 1
                        sum += img[i][j - 1]
                    }
                    if (j + 1 < column) {
                        count += 1
                        sum += img[i][j + 1]
                    }
                    if (i - 1 >= 0) {
                        count += 1
                        sum += img[i - 1][j]
                        if (j - 1 >= 0) {
                            count += 1
                            sum += img[i - 1][j - 1]
                        }
                        if (j + 1 < column) {
                            count += 1
                            sum += img[i - 1][j + 1]
                        }
                    }
                    if (i + 1 < row) {
                        count += 1
                        sum += img[i + 1][j]
                        if (j - 1 >= 0) {
                            count += 1
                            sum += img[i + 1][j - 1]
                        }
                        if (j + 1 < column) {
                            count += 1
                            sum += img[i + 1][j + 1]
                        }
                    }
                    result[i][j] = sum / count
                }
            }
            return result
        }
    }
}